mutacc/utils/vcf_handler.py

```diff
@@ -0,0 +1,212 @@
+"Module to handle vcf files"
+
+import logging
+
+import yaml
+
+LOG = logging.getLogger(__name__)
+
+
+class INFOParser:
+    """
+        Class to customize parsing of INFO column in vcf
+    """
+
+    def __init__(self, parser_info: list, stream: str):
+
+        if stream.lower() not in ("read", "write", "r", "w"):
+            LOG.warning("The stream must be set to either 'read' or 'write' ")
+        self.stream = stream.lower()[0]
+
+        if not self._check(parser_info):
+            LOG.warning("Parser info not given correctly")
+            raise ValueError
+
+        self.parsers = self._get_parsers(parser_info)
+
+    def parse(self, vcf_entry):
+        """
+            Given a vcf entry, parse the INFO column
+
+            args:
+                vcf_entry (Cyvcf2.Variant or dict)
+            returns:
+                results (dict): Dictionary with the parsed fields
+        """
+
+        if self.stream == "r":
+            results = {}
+            for parser in self.parsers:
+                if vcf_entry.INFO.get(parser["id"]):
+                    info_id = parser["id"]
+                    display_name = parser["display_name"]
+                    parser_func = parser["parse_func"]
+                    results[display_name] = parser_func(vcf_entry.INFO.get(info_id))
+            return results
+
+        results = []
+        for parser in self.parsers:
+            if vcf_entry.get(parser["id"]):
+                info_id = parser["id"]
+                parser_func = parser["parse_func"]
+                results.append(parser_func(vcf_entry[info_id]))
+        return ";".join(results)
+
+    def _get_parsers(self, parse_info):
```

Comment:
Consider naming and generalization depending on stream direction and what to call the ID